### DISCUSSION TITLE
Unit 3 - MyDataset vs TensorDataset

### DISCUSSION THREAD

**Discussion by pavilaalo** [Lightning-AI/dl-fundamentals#85] [Lecture videos and quizzes]
In unit 3 a custom class "MyDataset" was created. Is it similar to TensorDataset from torch.utils.data? any relevant differences?

Thanks

**Comment by rasbt** [comment]
Good question. Yes, they are relatively similar. The `MyDataset` is mainly to illustrate how to create a custom dataset. The small difference is also that it converts a NumPy array into a PyTorch tensor whereas `TensorDataset` expects tensors as input. But yeah, overall it's the same idea.

For those reading this, here are the code references:

- [TensorDataset](https://pytorch.org/docs/stable/_modules/torch/utils/data/dataset.html#TensorDataset)
- [MyDataset in Unit 3](https://github.com/Lightning-AI/dl-fundamentals/blob/main/unit03-pytorch-training/3.6-logreg-in-pytorch/logreg-part3.ipynb)
